src/webgpu/gpu_test.ts

```diff
@@ -1467,14 +1500,27 @@ export function TextureTestMixin<F extends FixtureClass<GPUTest>>(
                   resource: texture.createView({
                     baseMipLevel: mipLevel,
                     mipLevelCount: 1,
-                    baseArrayLayer: layer,
-                    arrayLayerCount: 1,
-                    dimension: '2d',
+                    ...(!this.isCompatibility && {
+                      baseArrayLayer: layer,
+                      arrayLayerCount: 1,
+                    }),
+                    dimension: pipelineType as GPUTextureViewDimension,
                   }),
```

Comment:
nit: this whole createView call is getting quite large to be inlined here, move it out to a separate variable